src/Workspaces/Remote/ServiceHub/CodeLensOOP/ReferenceCodeLensProvider.cs

```diff
@@ -0,0 +1,228 @@
+                        {
+                            var moniker = Util.GetImageMoniker(referenceLocationDescriptor.Glyph.Value);
+                            imageId = new ImageId(moniker.Guid, moniker.Id);
+                        }
+
+                        return new CodeLensDetailEntryDescriptor()
+                        {
+                            // use default since reference codelens don't require special behaviors
+                            NavigationCommand = null,
+                            NavigationCommandArgs = null,
+                            Tooltip = null,
+                            Fields = new List<CodeLensDetailEntryField>()
+                            {
+                                // file path
+                                new CodeLensDetailEntryField() { Text =   Descriptor.FilePath },
+                                // line number
+                                new CodeLensDetailEntryField() { Text = referenceLocationDescriptor.LineNumber.ToString() },
+                                // column number
+                                new CodeLensDetailEntryField() { Text = referenceLocationDescriptor.ColumnNumber.ToString() },
+                                // reference text
+                                new CodeLensDetailEntryField() { Text = referenceLocationDescriptor.ReferenceLineText },
+                                // reference start
+                                new CodeLensDetailEntryField() { Text = referenceLocationDescriptor.ReferenceStart.ToString() },
+                                // reference end
+                                new CodeLensDetailEntryField() { Text = (referenceLocationDescriptor.ReferenceStart + referenceLocationDescriptor.ReferenceLength).ToString() },
+                                // reference long description
+                                new CodeLensDetailEntryField() { Text = referenceLocationDescriptor.LongDescription },
+                                // reference image id
+                                new CodeLensDetailEntryField() { ImageId = imageId }, 
+                                // text before reference 2
+                                new CodeLensDetailEntryField() { Text = referenceLocationDescriptor.BeforeReferenceText2 },
+                                // text before reference 1
+                                new CodeLensDetailEntryField() { Text = referenceLocationDescriptor.BeforeReferenceText1 },
+                                // text after reference 1
+                                new CodeLensDetailEntryField() { Text = referenceLocationDescriptor.AfterReferenceText1 },
+                                // text after reference 2
+                                new CodeLensDetailEntryField() { Text = referenceLocationDescriptor.AfterReferenceText2 }
+                            },
+                        };
+                    }).ToList(),
```

Comment:
SelectAsArray...ToList?  Why not just Select....ToList?

Reply:
sure.